src/Utilities/QuaternionHelpers.hpp

```diff
@@ -0,0 +1,33 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines helper functions for converting between `std::array` and boost
+/// quaternions.
+
+#pragma once
+
+#include <boost/math/quaternion.hpp>
+
+template <typename T>
+std::array<T, 4> boostquat_to_array(const boost::math::quaternion<T>& input) {
```

Comment:
I suggest `quaternion_to_array`.  We're unlikely to have any other types of quaternions, and, if we do choose to switch implementations, using "boost" in the function names will make transitioning harder.  (Same for other functions.)